### PR TITLE
Create cluster by profile env - add parser from yaml to cluster obj

### PR DESCRIPTION
**Pull request by shellymiron**
Adding the option to create cluster using profile env from the chosen profiles in the `tf-cluster-profile.yml`:
- rosa-sts-pl
- rosa-sts-ad
- rosa-sts-upg

& parse profile yaml to cluster obj

### CODE REVIEW

File: tests/ci/file.go [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+	if err != nil {
+		f.Close()
+	}
+	return
+}
+
+// Rename Will rename the file from oldName to newName under folder typ
+func Rename(typ string, oldName string, newName string) (err error) {
+	// workaround for ROSA cluster
+	if typ == CON.MOAPlan {
```
Comment (amalykhi) [inline tests/ci/file.go:144]: what is moaplan?

File: tests/e2e/cluster_creation.go [hunk 2]
```diff
@@ -50,7 +51,7 @@ var _ = Describe("TF Test", func() {
 
 			By("Create account-roles")
 			accRoleParam := &EXE.AccountRolesArgs{
-				Token:             token,
+				Token:             os.Getenv(CON.TokenENVName),
```
Comment (amalykhi) [inline tests/e2e/cluster_creation.go:54]: I would take it once and save as constant/var

File: tests/e2e/cluster_creation.go [hunk 3]
```diff
@@ -59,7 +60,7 @@ var _ = Describe("TF Test", func() {
 
 			By("Create Cluster")
 			clusterParam := &EXE.ClusterCreationArgs{
-				Token:                token,
+				Token:                os.Getenv(CON.RhcsClusterProfileENV),
```
Comment (amalykhi) [inline tests/e2e/cluster_creation.go:63]: Do you mean  CON.TokenENVName here?

File: tests/utils/constants/frame_constants.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+	RecordFile               = "cluster.ini"
+	ResourceFile             = "resource.ini"
+	ProfileFile              = "profile.ini"
+	LockSuffix               = "lock"
+	// OSDSymbol, OCPSymbol, RHMISymbol ROSASymbol are the types of cluster, also used as subfolders to store the related
+	// clusters information under ClustersParentFolderName.
+	OSDSymbol                   = "OSD"
+	OCPSymbol                   = "OCP"
+	RHMISymbol                  = "RHMI"
+	MOASymbol                   = "MOA"
```
Comment (amalykhi) [inline tests/utils/constants/frame_constants.go:16]: MOA?

File: tests/utils/constants/frame_constants.go [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+	ClustersParentFolderName = ".datainfo"
+	RecordFile               = "cluster.ini"
+	ResourceFile             = "resource.ini"
+	ProfileFile              = "profile.ini"
+	LockSuffix               = "lock"
+	// OSDSymbol, OCPSymbol, RHMISymbol ROSASymbol are the types of cluster, also used as subfolders to store the related
+	// clusters information under ClustersParentFolderName.
+	OSDSymbol                   = "OSD"
+	OCPSymbol                   = "OCP"
+	RHMISymbol                  = "RHMI"
```
Comment (amalykhi) [inline tests/utils/constants/frame_constants.go:15]: what is RHMI?

File: tests/utils/constants/subscriptions.go [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+)
+
+func SubscriptionSystemUnits() []string {
+	return []string{
+		SystemUnitsCoresVCPU,
+		SystemUnitsSockets,
+	}
+}
+
+// ClusterLog Severity
```
Comment (amalykhi) [inline tests/utils/constants/subscriptions.go:205]: Could we set cluster log severity in terraform -provider @bardielle ?

File: tests/e2e/cluster_creation.go (hunk 2 above)
Comment (shellymiron) [reply tests/e2e/cluster_creation.go]: ack, changed to:
```
var (
	TokenENVName = os.Getenv("RHCS_TOKEN")
)
```

File: tests/ci/profiles/tf_cluster_profile.yml [hunk 7]
```diff
@@ -44,7 +45,7 @@
     proxy: true
     labeling: true
     tagging: true
-    channel_group: staging
+    channel_group: stable
```
Comment (amalykhi) [inline tests/ci/profiles/tf_cluster_profile.yml:48]: why stable. We want to check the releases ASAP

File: tests/ci/profile_handler.go [hunk 8]
```diff
@@ -109,13 +111,16 @@ func GenerateClusterCreationArgsByProfile(profile *Profile) (clusterArgs *EXE.Cl
 		acctPrefix := clusterArgs.ClusterName
 		majorVersion := ""
 		if profile.Version != "" {
-			majorVersion = strings.Join(strings.Split(profile.Version, ".")[0:2], ".")
+			if profile.Version != "latest" {
+				majorVersion = strings.Join(strings.Split(profile.Version, ".")[0:2], ".")
+			} else {
+				majorVersion = profile.Version
```
Comment (amalykhi) [inline tests/ci/profile_handler.go:117]: if version is specified as "latest" - major version should be retrieved by rhcs_versions  or another function, no?

File: tests/utils/constants/frame_constants.go [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+		OCPSymbol,
+		ROSASymbol,
+		ConfigSymbol,
+		OSDTrialSymbol,
+	}
+}
+
+const (
+	// Default properties for clusters
+	QEUsage = "qe_usage"
```
Comment (amalykhi) [inline tests/utils/constants/frame_constants.go:47]: What is the usage of QEUsage property?

File: tests/utils/constants/subscriptions.go [hunk 10]
```diff
@@ -0,0 +1,304 @@
...
+		RHOSAK,
+		RHOSAKTrial,
+		RHOSR,
+		RHOSRTrial,
+		RHACS,
+		RHOSE,
+	}
+}
+
+// Subscription Provenance, which is also Cluster Provenance.
```
Comment (amalykhi) [inline tests/utils/constants/subscriptions.go:49]: Is it relevant for tf-provider?

File: tests/ci/profiles/tf_cluster_profile.yml (hunk 7 above)
Comment (shellymiron) [reply tests/ci/profiles/tf_cluster_profile.yml:48]: there is no channel_group named "staging", only stable

Comment (amalykhi) [reply tests/ci/profiles/tf_cluster_profile.yml:48]: Sorry, I've meant "candidate"

Comment (shellymiron) [reply tests/ci/profiles/tf_cluster_profile.yml]: changed to candidate

File: tests/ci/config.go [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+	ClusterHomeDir string `env:"GSE_OUTPUT_DIR" default:"user.HomeDir" yaml:"clusterHomeDir,omitempty"`
+
+	// QuotaRuleDir is the default cloud provider used to provision cluster.
+	QuotaRuleDir string `env:"QUOTA_RULE_DIR" yaml:"quotaRuleDir,omitempty"`
+
+	// ChannelGroup is the channel where to get the openshift version.
+	ChannelGroup string `env:"CHANNEL_GROUP" yaml:"channel_group,omitempty"`
+
+	// OpenshiftVersion is the openshift version used on the cluster provision.
+	OpenshiftVersion string `env:"OPENSHIFT_VERSION" yaml:"openshift_version,omitempty"`
```
Comment (xueli181114) [inline tests/ci/config.go:45]: No need here

File: tests/ci/config.go [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+	DefaultClusterSymbol string `env:"DEFAULT_CLUSTER_SYMBOL" default:"OSD" yaml:"clusterSymbol,omitempty"`
+
+	// ClusterHomeDir is the default cloud provider used to provision cluster.
+	ClusterHomeDir string `env:"GSE_OUTPUT_DIR" default:"user.HomeDir" yaml:"clusterHomeDir,omitempty"`
+
+	// QuotaRuleDir is the default cloud provider used to provision cluster.
+	QuotaRuleDir string `env:"QUOTA_RULE_DIR" yaml:"quotaRuleDir,omitempty"`
+
+	// ChannelGroup is the channel where to get the openshift version.
+	ChannelGroup string `env:"CHANNEL_GROUP" yaml:"channel_group,omitempty"`
```
Comment (xueli181114) [inline tests/ci/config.go:42]: deprecated in CMS

File: tests/ci/config.go [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+	DefaultCloudProvider string `env:"DEFAULT_CLOUD_PROVIDER" default:"aws" yaml:"cloudProvider,omitempty"`
+
+	// DefaultClusterSymbol is the default type of the cluster. The type decides the product id of the cluster.
+	DefaultClusterSymbol string `env:"DEFAULT_CLUSTER_SYMBOL" default:"OSD" yaml:"clusterSymbol,omitempty"`
+
+	// ClusterHomeDir is the default cloud provider used to provision cluster.
+	ClusterHomeDir string `env:"GSE_OUTPUT_DIR" default:"user.HomeDir" yaml:"clusterHomeDir,omitempty"`
+
+	// QuotaRuleDir is the default cloud provider used to provision cluster.
+	QuotaRuleDir string `env:"QUOTA_RULE_DIR" yaml:"quotaRuleDir,omitempty"`
```
Comment (xueli181114) [inline tests/ci/config.go:39]: I don't think we need this. We don't need to cover quota

File: tests/ci/config.go [hunk 14]
```diff
@@ -0,0 +1,198 @@
...
+	// QEFlag is the default property of the cluster created by the script, etc. the test run id.
+	QEFlag         string `env:"QE_FLAG" default:"" yaml:"qeFlag,omitempty"`
+	QEUsage        string `env:"QE_USAGE" default:"" yaml:"qeUsage,omitempty"`
+	DeploymentName string `env:"DEPLOYMENT_NAME" default:"main" yaml:"deploymentName,omitempty"`
+
+	// DefaultCloudProvider is the default cloud provider used to provision cluster.
+	DefaultCloudProvider string `env:"DEFAULT_CLOUD_PROVIDER" default:"aws" yaml:"cloudProvider,omitempty"`
+
+	// DefaultClusterSymbol is the default type of the cluster. The type decides the product id of the cluster.
+	DefaultClusterSymbol string `env:"DEFAULT_CLUSTER_SYMBOL" default:"OSD" yaml:"clusterSymbol,omitempty"`
```
Comment (xueli181114) [inline tests/ci/config.go:33]: No need

File: tests/ci/config.go [hunk 15]
```diff
@@ -0,0 +1,198 @@
...
+	// The supported values for FakeCluster are 'fake', 'mixed', 'real'.
+	FakeCluster bool `env:"FAKE_CLUSTER" default:"mixed" yaml:"fake_cluster,omitempty"`
+
+	// Debug model
+	Debug bool `env:"DEBUG" default:"true" yaml:"debug"`
+
+	// QEFlag is the default property of the cluster created by the script, etc. the test run id.
+	QEFlag         string `env:"QE_FLAG" default:"" yaml:"qeFlag,omitempty"`
+	QEUsage        string `env:"QE_USAGE" default:"" yaml:"qeUsage,omitempty"`
+	DeploymentName string `env:"DEPLOYMENT_NAME" default:"main" yaml:"deploymentName,omitempty"`
```
Comment (xueli181114) [inline tests/ci/config.go:27]: No need

File: tests/ci/config.go [hunk 16]
```diff
@@ -0,0 +1,198 @@
...
+	RHCSEnv string `env:"RHCS_ENV" default:"stage" yaml:"env"`
+
+	// The supported values for FakeCluster are 'fake', 'mixed', 'real'.
+	FakeCluster bool `env:"FAKE_CLUSTER" default:"mixed" yaml:"fake_cluster,omitempty"`
+
+	// Debug model
+	Debug bool `env:"DEBUG" default:"true" yaml:"debug"`
+
+	// QEFlag is the default property of the cluster created by the script, etc. the test run id.
+	QEFlag         string `env:"QE_FLAG" default:"" yaml:"qeFlag,omitempty"`
```
Comment (xueli181114) [inline tests/ci/config.go:25]: No need

File: tests/ci/config.go [hunk 17]
```diff
@@ -0,0 +1,198 @@
...
+
+var RHCS = new(RHCSconfig)
+
+// RHCSConfig contains platforms info for the RHCS testing
+type RHCSconfig struct {
+	// Env is the OpenShift Cluster Management environment used to provision clusters.
+	RHCSEnv string `env:"RHCS_ENV" default:"stage" yaml:"env"`
+
+	// The supported values for FakeCluster are 'fake', 'mixed', 'real'.
+	FakeCluster bool `env:"FAKE_CLUSTER" default:"mixed" yaml:"fake_cluster,omitempty"`
```
Comment (xueli181114) [inline tests/ci/config.go:19]: No need

File: tests/ci/config.go [hunk 18]
```diff
@@ -0,0 +1,198 @@
...
+	"path"
+	"strings"
+)
+
+var RHCS = new(RHCSconfig)
+
+// RHCSConfig contains platforms info for the RHCS testing
+type RHCSconfig struct {
+	// Env is the OpenShift Cluster Management environment used to provision clusters.
+	RHCSEnv string `env:"RHCS_ENV" default:"stage" yaml:"env"`
```
Comment (xueli181114) [inline tests/ci/config.go:16]: Default value set to staging

File: tests/ci/config.go [hunk 19]
```diff
@@ -0,0 +1,198 @@
...
+	ProductID           string `env:"PRODUCT_ID" yaml:"product_id,omitempty"`
+	AWSCredRequestsPath string `env:"AWS_CREDREQUESTS_PATH" yaml:"awsCredRequestsPath,omitempty"`
+	AWSTemplates        string `env:"AWS_TEMPLATES_PATH" yaml:"awsTemplatesPath,omitempty"`
+	ClusterProfile      string `env:"CLUSTER_PROFILE" yaml:"clusterProfile,omitempty"`
+	ClusterProfileDir   string `env:"CLUSTER_PROFILE_DIR" yaml:"clusterProfileDir,omitempty"`
+
+	OutputDir         string
+	ProfileConfigsDir string
+	YAMLProfilesDir   string
+	VPCFolder         string
```
Comment (xueli181114) [inline tests/ci/config.go:63]: No need

File: tests/ci/config.go [hunk 20]
```diff
@@ -0,0 +1,198 @@
...
+	if os.Getenv("QE_FLAG") != "" {
+		RHCS.QEFlag = os.Getenv("QE_FLAG")
+	}
+
+	RHCS.QEUsage = "ocm-api-testing"
+	if os.Getenv("QE_USAGE") != "" {
+		RHCS.QEUsage = os.Getenv("QE_USAGE")
+	}
+
+	RHCS.DeploymentName = "main"
```
Comment (xueli181114) [inline tests/ci/config.go:109]: Remove the no need env settings

File: tests/ci/file.go [hunk 21]
```diff
@@ -0,0 +1,269 @@
...
+	CON "github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/constants"
+	EXE "github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/exec"
+	f "github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/files"
+	h "github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/helper"
+)
+
+// The cfg will be used to define the testing environment
+var cfg = RHCS
+
+func HomeDir() string {
```
Comment (xueli181114) [inline tests/ci/file.go:22]: We don't need to record file on local. Terraform will record files on local. Remove all records part. And we don't need to do e2e for now. Just SDK is required.

File: tests/ci/file.go [hunk 22]
```diff
@@ -0,0 +1,269 @@
...
+			continue
+		}
+		err = os.MkdirAll(dir, os.ModePerm)
+	}
+
+	return
+}
+
+// PrepareAWSAccessKeysDir will create the directories of the AWS access keys
+func PrepareAWSAccessKeysRecordFile() (err error) {
```
Comment (xueli181114) [inline tests/ci/file.go:129]: No need either

File: tests/ci/profile_handler.go [hunk 23]
```diff
@@ -88,8 +89,16 @@ func PrepareRoute53() {}
 
 func GenerateClusterCreationArgsByProfile(profile *Profile) (clusterArgs *EXE.ClusterCreationArgs, manifestsDir string, err error) {
 	clusterArgs = &EXE.ClusterCreationArgs{
-		Token: os.Getenv(CON.TokenENVName),
+		Token: CON.TokenENVName,
 	}
+
+	if CON.ManifestsDirENV == "" {
+		// deafult profile dir selected
+		profile.ManifestsDIR = CON.ROSAClassic
```
Comment (xueli181114) [inline tests/ci/profile_handler.go:97]: ManifestsDirENV is used to set the whole manifest dir path. Please don't set it to rosaclassic

File: tests/utils/constants/frame_constants.go [hunk 24]
```diff
@@ -0,0 +1,65 @@
...
+	AWSAccessKeysFile     = ".aws/credentials"
+	StackTemplateFilePath = ""
+	TFYAMLProfile         = "tf_cluster_profile.yml"
+
+	// SECTOR is the sector in ROSA/OSD env
+	Sector = "SECTOR"
+)
+
+// Symbols returns a list of the symbols. If a new folder is wanted to created in .datainfo, the symbol should be added in the list
+func Symbols() []string {
```
Comment (xueli181114) [inline tests/utils/constants/frame_constants.go:35]: No need for Symbols

File: tests/utils/constants/frame_constants.go [hunk 25]
```diff
@@ -0,0 +1,65 @@
...
+	// ClustersParentFolderName is the folder name to store the cluster
+	ClustersParentFolderName = ".datainfo"
+	RecordFile               = "cluster.ini"
+	ResourceFile             = "resource.ini"
+	ProfileFile              = "profile.ini"
+	LockSuffix               = "lock"
+	// OSDSymbol, OCPSymbol, RHMISymbol ROSASymbol are the types of cluster, also used as subfolders to store the related
+	// clusters information under ClustersParentFolderName.
+	OSDSymbol                = "OSD"
+	OCPSymbol                = "OCP"
```
Comment (xueli181114) [inline tests/utils/constants/frame_constants.go:14]: No need OCP related

**Comment by xueli181114** [comment]
@shellymiron Let's don't copy backend part directly. Backend is too complicated which supports AMS,CMS,FleetManager automation testing. So let's clean the used part, just pick the things we needed.

File: tests/ci/config.go [hunk 26]
```diff
@@ -0,0 +1,198 @@
...
+	RHCS.OutputDir = path.Join(rootDir, "output")
+	RHCS.VPCFolder = path.Join(RHCS.OutputDir, "vpcs")
+
+	RHCS.RHCSEnv = "stage"
+	if os.Getenv("OCM_ENV") != "" {
+		RHCS.RHCSEnv = os.Getenv("OCM_ENV")
+		if RHCS.RHCSEnv == "prod" {
+			RHCS.RHCSEnv = "production"
+		}
+	}
```
[the remark this answers is not in the document]
Comment (shellymiron) [reply tests/ci/config.go]: done :)

File: tests/ci/config.go [hunk 27]
```diff
@@ -0,0 +1,51 @@
...
+
+	YAMLProfilesDir string
+}
+
+func init() {
+	currentDir, _ := os.Getwd()
+	project := "terraform-provider-rhcs"
+	rootDir := GetEnvWithDefault(CON.WorkSpace, strings.SplitAfter(currentDir, project)[0])
+
+	RHCS.RHCSEnv = "staging"
```
Comment (amalykhi) [inline tests/ci/config.go:29]: Why do you set it here hardcoded? It is the same with default value

File: tests/ci/config.go [hunk 28]
```diff
@@ -0,0 +1,51 @@
...
+
+	CON "github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/constants"
+)
+
+var RHCS = new(RHCSconfig)
+
+// RHCSConfig contains platforms info for the RHCS testing
+type RHCSconfig struct {
+	// Env is the OpenShift Cluster Management environment used to provision clusters.
+	RHCSEnv           string `env:"RHCS_ENV" default:"staging" yaml:"env"`
```
Comment (amalykhi) [inline tests/ci/config.go:17]: How would I change it to another environment? with environment variable from PROW CI job?

Comment (shellymiron) [reply tests/ci/config.go:17]: yeah with the `CON.OCMEnv` var declared

File: tests/ci/profile_handler.go (hunk 8 above)
Comment (shellymiron) [reply tests/ci/profile_handler.go]: fixed

File: tests/ci/config.go (hunk 27 above)
Comment (shellymiron) [reply tests/ci/config.go]: fixed

File: go.mod [hunk 29]
```diff
@@ -1,9 +1,10 @@
 module github.com/terraform-redhat/terraform-provider-rhcs
 
-go 1.18
+go 1.20
```
Comment (amalykhi) [inline go.mod:3]: Can you run 'go mod tidy' and commit the resulted changes?
it should fix the CI

Comment (shellymiron) [reply go.mod:3]: that's what I'm trying to do
seems to not work :/

Comment (shellymiron) [reply go.mod:3]: fixed by changing `check-pull-request.yaml` `go-version` to 1.20

File: tests/utils/files/ini_file.go [hunk 30]
```diff
@@ -0,0 +1,43 @@
+package files
+
+import (
+	"fmt"
+
+	"gopkg.in/ini.v1"
+)
+
+// IniConnection builds the connection of the ini file
+func IniConnection(filename string, create ...bool) *ini.File {
```
Comment (xueli181114) [inline tests/utils/files/ini_file.go:10]: We won't have ini file stored on local. No need to parse ini files

File: tests/utils/files/common.go [hunk 31]
```diff
@@ -0,0 +1,39 @@
+package files
```
Comment (xueli181114) [inline tests/utils/files/common.go:1]: All files not required. tf will record its local files

File: tests/utils/exec/cluster.go [hunk 32]
```diff
@@ -110,11 +113,9 @@ func NewClusterService(manifestDir string) *ClusterService {
 	return sc
 }
 
-//******************************************************
-
 func CreateTFCluster(ctx context.Context, manifestsDir string,
 	varArgs map[string]interface{}, abArgs ...string) (string, error) {
-	targetDir := CON.GrantClusterManifestDir(manifestsDir)
+	targetDir := CON.GrantClusterManifestDir(CON.ROSAClassic)
```
Comment (xueli181114) [inline tests/utils/exec/cluster.go:118]: manifestsDir is required parameter, and we will have multiple manifests dirs. please don't hardcode it to rosaclassic

File: tests/utils/constants/frame_constants.go [hunk 33]
```diff
@@ -0,0 +1,29 @@
+package constants
```
Comment (xueli181114) [inline tests/utils/constants/frame_constants.go:1]: All constants no needed except TFYAMLProfile

**Comment by bardielle** [comment]
@amalykhi can you approve it? or add more comments

File: tests/README.md [hunk 34]
```diff
@@ -51,14 +51,19 @@ The cluster created by the automation scripts are shared across all of the test
 ### Prerequisite
 Please read repo's [README.md](../README.md)
 #### Users and Tokens
-To execute the test cases, we need to prepare the [offline token](https://console.redhat.com/openshift/token/show). Get the offline token and export as an ENV variable **rhcs_TF_TOKEN**.
-* export rhcs_TF_TOKEN=<offline token>
+To execute the test cases, we need to prepare the [offline token](https://console.redhat.com/openshift/token/show). Get the offline token and export as an ENV variable 
 
+**rhcs_TF_TOKEN**.
+* export RHCS_TOKEN=<offline token>
+* export SDQE_ROSA_API=<offline token> (temporarily)
```
Comment (amalykhi) [inline tests/README.md:58]: Why do we need additional token?

Comment (shellymiron) [reply tests/README.md:58]: for this:
`versionsList := HELPER.GetGreaterOrEqualVersions(RosaConnection, majorVersion, profile.ChannelGroup, true, true)`

File: tests/utils/constants/http_constants.go [hunk 35]
```diff
@@ -0,0 +1,21 @@
+package constants
+
+// HTTP response code
+const (
+	HTTPOK        = 200
```
Comment (amalykhi) [inline tests/utils/constants/http_constants.go:5]: Is it used?
import "net/http"
it includes all the statuses , i.e. http.StatusOK

File: tests/README.md (hunk 34 above)
Comment (shellymiron) [reply tests/README.md:58]: the `RHCS_TOKEN` is a string, while this `SDQE_ROSA_API` is a connection obj

Comment (shellymiron) [reply tests/README.md]: oh, you are right
I fixed that :)

**Comment by amalykhi** [comment]
@shellymiron Please, rebase. I noticed where are could be duplications of code in "connection" part because of my PR. Feel free to move functions if needed

**Comment by shellymiron** [comment]
image

successful e2e run

**Comment by xueli181114** [comment]
@shellymiron Is there any other blockers for this PR merging?

File: tests/README.md [hunk 36]
```diff
@@ -51,14 +51,18 @@ The cluster created by the automation scripts are shared across all of the test
...
+**rhcs_TF_TOKEN**.
+* export RHCS_TOKEN=<offline token>
 
 #### Global variables
 The default rhcs enviroment is `production`. To specify running on staging or local, can export the global vairable 
-* export GATE_WAY="https://api.stage.openshift.com"
+* export GATE_WAY = "https://api.stage.openshift.com"
+* export RHCSEnv = "staging"
 
+To decalre the profile cluster type to be run, use the below var:
```
Comment (amalykhi) [inline tests/README.md:64]:
```suggestion
To declare the profile cluster type to be run, use the below var:
```

**Pull request merged**